rfc/reproducible-builds.md

```diff
@@ -0,0 +1,66 @@
+# Reproducible Builds
+To ensure the security of constellation's supply chain, we need to make our software builds reproducible.
+This is the only way to ensure a verifiable path from source code to binary.
+Every step of the build process must be deterministic.
+
+## Definition of Goals
+1. All our OCIs executed in the cluster must be reproducible bit by bit .
+1. For that, every compiled executable has to be deterministically compiled.
+1. If we change parts of our codebase (i.e. add several other programming languages) we should not have to make major changes to the image build system.
+1. Moving from `Docker` to a reproducible build system should have a minimum overhead. Ideally, we can reuse our existing `Dockerfile`s.
+1. The tool that builds the OCIs should be battle proven and reliable.
+1. The image should be lightweight.
+
+## Steps to Achieve Goals
+
+### Executables
+* Remove metadata from binaries, namely:
+  * Timestamps
+  * Build IDs
+  * Symbols
+  * Version numbers
+  * Paths
+  * Anything that might change for another host OS / time / ...
+* Eliminate dependencies on libraries (make executable static)
+
+Striping metadata from the binary can be done in the building process.
+This can be achieved by setting the appropriate compiler flags.
+
+```bash
+$ CGO_ENABLED=0 go build -o <out_name> -buildvcs=false -trimpath -ldflags "-s -w -buildid=''"
```

Comment:
Maybe we could summarize shortly (or link a resource that does) what these flags achieve. I had to look up the meaning of `-s -w`, but `go tool link` provides the information.